Error: Failed to parse given uname release: 4.19.13-1901141830.iot-lts2018-sos when updating ACRN setup

LOG from the update:   [acrn_swupd_update.txt](https://github.com/clearlinux/swupd-client/files/2842695/acrn_swupd_update.txt)

**Description:** I am updating a previously configured ACRN environment on an apolo lake nuc. The swupd update does not fail, but it does show Errors involving the kernel being run. 

I opened this issue because I cannot determine if its an error or expected. But it does intrigue me since its the first time I am seing them. I have gone through the update procedure with ACRN and Clear linux since forever and this is the first time this errors are present.

Hardware specifications are described in the attached file ACRN_hardware_specifications.txt
[ACRN_hardware_specifications.txt](https://github.com/clearlinux/swupd-client/files/2842715/ACRN_hardware_specifications.txt)

The steps to setup a NUC with the ACRN can be found in this wiki:  https://projectacrn.github.io/latest/getting-started/apl-nuc.html

I was wrong. I now have a system that does **not** have any of those older kernels with the broken release field. But I am getting the same error as reported by @LibertadCruz. I actually bumped into this when installing a new kernel bundle. I only see this when running on top of ACRN (so running the `iot-lts2018-sos` kernel). If I reboot the exact same system and run Clear Linux natively, I do not get any error.

I should note, in case you do not know those details, that booting under ACRN is slightly different and while `systemd-boot` is used, a special entry (`acrn.conf`) is added manually - see https://projectacrn.github.io/latest/getting-started/apl-nuc.html#add-the-acrn-hypervisor-to-the-efi-partition for more details. I do not know whether this matters or not.

Here is the output in both scenarii:

### Running on top of ACRN
```
$ sudo swupd bundle-add kernel-native
Downloading packs...
	...100%
Starting download of remaining update content. This may take a while...
	...100%
Finishing download of update content...
Installing bundle(s) files...
	...100%
Calling post-update helper scripts.
none
[ERROR] cbm (../src/bootman/bootman.c:L681): Failed to parse given uname release: 4.19.19-9.iot-lts2018-sos
[WARNING] cbm (../src/bootman/bootman.c:L70): Unable to parse the currently running kernel: 4.19.19-9.iot-lts2018-sos
[ERROR] cbm (../src/bootman/update.c:L196): Cannot determine the currently running kernel
Successfully installed 1 bundle
gvancuts@cs-corei5 ~ $ uname -r
4.19.19-9.iot-lts2018-sos
```

### Running native Clear Linux
```
$ sudo swupd bundle-add kernel-native
Downloading packs...
	...100%
Starting download of remaining update content. This may take a while...
	...100%
Finishing download of update content...
Installing bundle(s) files...
	...100%
Calling post-update helper scripts.
none
Successfully installed 1 bundle
gvancuts@cs-corei5 ~ $ uname -r
4.19.20-420.lts
```

I'm running clr-boot-manager version 3.1.0:
```
$ clr-boot-manager version
clr-boot-manager - version 3.1.0

Copyright © 2016-2017 Intel Corporation

clr-boot-manager is free software; you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation; either version 2.1 of the License, or
(at your option) any later version.
```

--- a/src/bootman/bootman.h
+++ b/src/bootman/bootman.h
@@ -32,7 +32,7 @@ typedef enum {
 /**
  * Maximum length for each component in a kernel identifier
  */
-#define CBM_KELEM_LEN 15
+#define CBM_KELEM_LEN 31
 
 /**
  * Represents the currently running system kernel

--- a/src/bootman/kernel.c
+++ b/src/bootman/kernel.c
@@ -449,7 +449,7 @@ bool cbm_parse_system_kernel(const char *inp, SystemKernel *kernel)
                 ++len;
         }
 
-        if (len < 1 || len >= CBM_KELEM_LEN) {
+        if (len < 1 || len > CBM_KELEM_LEN) {
                 return false;
         }
 

